pkg/frontend/transport/handler.go

```diff
@@ -77,8 +79,27 @@ func NewHandler(cfg HandlerConfig, roundTripper http.RoundTripper, log log.Logge
 			Help: "Total amount of wall clock time spend processing queries.",
 		}, []string{"user"})
 
+		// Empty objectives for these summaries on purpose since they can't be aggregated
+		// and so we are just using them for the convenience of sum and count metrics. No
+		// histograms here since the cardinality from the number of buckets required to
+		// understand query responses is prohibitively expensive.
+
+		h.querySeries = promauto.With(reg).NewSummaryVec(prometheus.SummaryOpts{
+			Name:       "cortex_query_series",
+			Help:       "Number of series returned by successful queries.",
```

Comment:
It's not the number of series returned as query response, but series fetched to execute the query.